src/Diagnostics/dump_spectra_gcm.jl

```diff
@@ -0,0 +1,145 @@
+
+struct SpectraGCMDiagnosticsParams <: DiagnosticsGroupParams
+    nor::Float64
+end
+
+"""
+    setup_dump_gcm_spectra_diagnostics(
+        ::ClimateMachineConfigType,
+        interval::String,
+        out_prefix::String;
+        writer = NetCDFWriter(),
+        interpol = nothing,
+        nor = Inf,
+    )
+
+Create and return a `DiagnosticsGroup` containing a diagnostic that
+dumps the spectrum at the specified
+`interval` after the velocity fields have been interpolated, into NetCDF files prefixed by
+`out_prefix`.
+"""
+function setup_dump_gcm_spectra_diagnostics(
+    ::ClimateMachineConfigType,
+    interval::String,
+    out_prefix::String;
+    writer = NetCDFWriter(),
+    interpol = nothing,
+    nor = Float64(1),
+)
+    @assert !isnothing(interpol)
+
+    return DiagnosticsGroup(
+        "Spectra",
+        Diagnostics.dump_gcm_spectra_init,
+        Diagnostics.dump_gcm_spectra_fini,
+        Diagnostics.dump_gcm_spectra_collect,
+        interval,
+        out_prefix,
+        writer,
+        interpol,
+        SpectraGCMDiagnosticsParams(nor),
+    )
+end
+
+function get_gcm_spectra(mpicomm, mpirank, Q, bl, interpol, nor)
+    FT = eltype(Q)
+    device = array_device(Q)
+    if device isa CPU
+        ArrayType = Array
+    else
+        ArrayType = CUDA.CuArray
+    end
+
+    istate =
+        ArrayType{FT}(undef, interpol.Npl, number_states(bl, Prognostic()))
+    interpolate_local!(interpol, Q.realdata, istate)
+
+    # TODO: get indices here without hard-coding them
+    _ρu, _ρv, _ρw = 2, 3, 4
+    project_cubed_sphere!(interpol, istate, (_ρu, _ρv, _ρw))
+
+    all_state_data = accumulate_interpolated_data(mpicomm, interpol, istate)
+
+    if mpirank == 0
+        #u = all_state_data[:, :, :, 2] ./ all_state_data[:, :, :, 1]
+        var_grid = all_state_data[:, :, :, 3] ./ all_state_data[:, :, :, 1]
+        dims = dimensions(interpol)
+        lat = dims["lat"][1]
+        lon = dims["long"][1]
+        level = dims["level"][1]
+        mass_weight = ones(FT, length(level)); # TODO: interpolate on pressure levs and do mass weighted calculations
+        spectrum_1d, m = power_spectrum_zonal( var_grid, level, lat, lon, mass_weight )
+        spectrum_2d, m_and_n, _ , __ = power_spectrum_gcm_2d( var_grid, mass_weight )
+        return spectrum_1d, m, spectrum_2d, m_and_n
+    end
+    return nothing, nothing
+end
+
+function dump_gcm_spectra_init(dgngrp, currtime)
+    Q = Settings.Q
+    bl = Settings.dg.balance_law
+    mpicomm = Settings.mpicomm
+    mpirank = MPI.Comm_rank(mpicomm)
+    FT = eltype(Q)
+    interpol = dgngrp.interpol
+    nor = dgngrp.params.nor
+
+    # get the 1d and 2d spectra and their associated wavenumbers
+    spectrum_1d, m, spectrum_2d, m_and_n = get_gcm_spectra(mpicomm, mpirank, Q, bl, interpol, nor)
+    if mpirank == 0
+        # Setup dimensions 
+        interpol_dims = dimensions(interpol)
+        lat = interpol_dims["lat"][1]
+        level = interpol_dims["level"][1]
```

Comment:
I think this is the distance from the Earth center. Earth radius should be removed to get the altitude heights.

Reply:
@LenkaNovak: I added a fix for this [here](https://github.com/CliMA/ClimateMachine.jl/pull/1559/files#diff-5a9fe607558ecfe8ea9b7ac212edf2a66e2b89d34d5d075a2e2b178428787d20R103-R107), please verify?